src/test/java/org/candlepin/subscriptions/tally/job/CaptureSnapshotsTaskManagerTest.java

```diff
@@ -135,6 +148,33 @@ void ensureNoUpdatesWhenAccountListCanNotBeRetreived() throws Exception {
         verify(queue, never()).enqueue(any());
     }
 
+    /**
+     * Test hourly snapshots using the minutes offset.  The offset is designed to ensure the snapshot
+     * includes as many finished tallies as possible, and allows an additional hour for them to finish.
+     *
+     * @throws Exception
+     */
+    @Test
+    void testHourlySnapshotTallyOffset() throws Exception {
+        List<String> expectedAccounts = Arrays.asList("a1", "a2");
+        when(accountListSource.syncableAccounts()).thenReturn(expectedAccounts.stream());
+
+        Duration metricRange = appProperties.getMetricLookupRangeDuration();
+        Duration prometheusLatencyDuration = appProperties.getPrometheusLatencyDuration();
+        Duration hourlyTallyOffsetMinutes = appProperties.getHourlyTallyOffset();
+
+        OffsetDateTime endDateTime = adjustTimeForLatency(OffsetDateTime.now(applicationClock.getClock())
+            .minus(hourlyTallyOffsetMinutes).truncatedTo(ChronoUnit.HOURS), prometheusLatencyDuration);
+        OffsetDateTime startDateTime = endDateTime.minus(metricRange);
+
+        manager.updateHourlySnapshotsForAllAccounts();
+
+        expectedAccounts.forEach(accountNumber -> {
+            verify(queue, times(1)).enqueue(createHourlyDescriptor(accountNumber,
+                startDateTime.toString(), endDateTime.toString()));
+        });
```

Comment:
This is an example of how I'd prefer this test be implemented (there are other ways to do this I'd be fine with too, e.g. org.mockito.ArgumentCaptor). I realize it is different than other tests, but this intentional. I'm concerned with making sure we test the math is correct rather than repeat the math in the test. Otherwise a flaw in the code that does all the arithmetic will occur in both the test and in the code being tested :-)

```suggestion
        expectedAccounts.forEach(accountNumber -> {
            verify(queue, times(1)).enqueue(
                TaskDescriptor.builder(TaskType.UPDATE_HOURLY_SNAPSHOTS, taskQueueProperties.getTopic())
                .setSingleValuedArg("accountNumber", accountNumber)
                // 2019-05-24T12:35Z truncated to top of the hour - 4 hours prometheus latency - 1 hour
                // tally latency - 24 hour metric range
                .setSingleValuedArg("startDateTime", "2019-05-23T07:00Z")
                .setSingleValuedArg("endDateTime", "2019-05-24T07:00Z").build());
        });
```